node/src/reactor/joiner.rs

```diff
@@ -395,23 +395,40 @@ impl reactor::Reactor for Reactor {
 
         let init_hash = config.node.trusted_hash;
 
-        match init_hash {
+        let linear_chain_sync = match init_hash {
+            Some(hash) => {
+                info!("Synchronizing linear chain from: {:?}", hash);
+                LinearChainSync::sync_trusted_hash(hash)
+            }
             None => {
-                let genesis = &chainspec_loader.chainspec().genesis;
-                let era_duration = genesis.highway_config.era_duration;
-                if Timestamp::now() > genesis.timestamp + era_duration {
-                    error!(
-                        "Node started with no trusted hash after the expected end of \
+                // Try the last seen finalized block.
+                let lfb: Option<Block> = storage
+                    .get_highest_block()
+                    .map_err(|err| storage::Error::from(err))?;
+                match lfb {
+                    None => {
+                        let genesis = &chainspec_loader.chainspec().genesis;
+                        let era_duration = genesis.highway_config.era_duration;
```

Comment:
> We could take the maximum of this and the minimum era height times the minimum round length, like we do in the era supervisor. Not sure it's worth it.

I haven't change this part but yes, we _could_ do what you're describing although we agreed (:grimacing: )that we won't panic if it's past that period so I am not sure it's worth it.

> Also, in the long run, it should still be fine to start with genesis up to the unbonding delay, which is several times this value.

Yes, I haven't implemented the part that syncs descendants of the Genesis yet. That's a bit more difficult since it's not really a block and the `linear_chain_sync` logic downloads descendants if _a block_.

Reply:
Done in [`862d9ed` (#731)](https://github.com/CasperLabs/casper-node/pull/731/commits/862d9edb286a91e18e8d2981fed0d6a74835c3ae)

Reply:
Wrong thread? (Anyway, not a blocker from my side. :+1:)